tests/integration/linux_bridge_test.py

```diff
@@ -189,12 +190,31 @@ def test_add_port_to_existing_bridge(bridge0_with_port0, port1_up):
     assertlib.assert_state(desired_state)
 
 
-def test_linux_bridge_uses_the_port_mac(port0_up, bridge0_with_port0):
-    port0_name = port0_up[Interface.KEY][0][Interface.NAME]
-    current_state = show_only((TEST_BRIDGE0, port0_name))
-    assert_mac_address(
-        current_state, port0_up[Interface.KEY][0][Interface.MAC]
-    )
+@pytest.mark.xfail(
+    is_fedora(),
+    reason='Fedora 31+ does not use kernel MAC for bridge',
+    raises=AssertionError,
+    strict=True,
+)
+def test_linux_bridge_uses_the_port_mac(port0_up):
```

Comment:
IIRC, this test was meant to document the intended/expected behavior with bridges and MAC addresses with Nmstate. It does not seem to be right to just expect this test to fail in the future. There should be a clear idea what should happen in this situation, it could also be something like it is undefined behavior. In this case we can probably just remove this test.

Reply:
Changed to:

* `test_linux_bridge_uses_the_port_mac_implicitly`
Expected to pass on CentOS 8 but fail on Fedora.
* `test_linux_bridge_uses_the_port_mac_explicitly`
Expected to pass on all systems.

Reply:
Thank you! I do not yet follow what behavior do we want in an ideal Nmstate? Should it also use the port mac implicitly on Fedora? Then there should be a card/issue that is references in the xfail. Do we care that this behavior stays the same in CentOS 8? What if it will change for CentOS 9 (which is likely if it is different in Fedora).

Reply:
For people never said MAC address in desire state, that means they should not assume the choice of bridge MAC address. For people who care, do it explicitly.

The reason I add this test case is Edy said RHV is expecting MAC address choose by kernel
not systemd-udev and he want a test case tracking it.

Most likely, CentOS 9 will fail as Fedora did.

To instruct Fedora on using kernel MAC address:

Change `/usr/lib/systemd/network/99-default.link` to:

```
[Link]
NamePolicy=kernel database onboard slot path
MACAddressPolicy=none
```

Reply:
I see. If we want to endorse the explicit MAC address naming, we should somehow mention it more user visible IMHO. Maybe add a section to the README with special behavior and mention the systemd config to get the other behavior? Also the xfail for Fedora should say that this behavior is not going to be changed by Nmstate. It seems to me that skipping the test instead of xfailing it makes even more sense in this case.

Reply:
systemd decision here is not backward compatible and I see no real chance for it to enter RHEL/CentOS. It is also totally wrong IMO.
As @thom311 also mentioned, it is also racefull (with client changing it).